pkg/healthautoexport/types.go

```diff
@@ -415,12 +431,26 @@ type Time struct {
 	time.Time
 }
 
-func NewTime(t time.Time) *Time {
-	return &Time{Time: t}
+func NewTime(t time.Time) Time {
+	return Time{Time: t}
+}
+
+// ParseTime attempts to parses the input string in the time format expected by health auto export.
+func ParseTime(s string) (Time, error) {
+	var multiErr error
+	for _, timeFormat := range TimeFormats {
+		parsed, err := time.Parse(timeFormat, s)
+		if err != nil {
+			multiErr = multierror.Append(multiErr, errors.Wrapf(err, `cannot parse with format: "%v"`, timeFormat))
+			continue
+		}
+		return NewTime(parsed), nil
+	}
+	return Time{}, fmt.Errorf(`failed to parse time across all known formats "%v": %w`, s, multiErr)
```

Comment:
It's so that we don't have to escape the double quotes within the quotes. Otherwise it will be written as such:

```go
fmt.Errorf("failed to parse time \"%v\" across all known formats: %w", s, err)
```

Personally I'm just used to using backticks when I need to include double quotes in strings 😅

Reply:
Ah right. I guess I tend to use single quotes to avoid that 😀